### ISSUE TITLE
Adding $iterator to foreach bindingContext

### ISSUE THREAD

**Issue by grofit** [knockout/knockout#1697]
Currently when you access a custom binding within a foreach you get access to the $index, however there is no way to get access to the iterator. It is a rare use case but I was wondering if it should be discussed as including into the default foreach binding or not.

http://stackoverflow.com/questions/28517590/knockout-custom-binding-in-foreach-to-get-itterator
Here is the post that started the discussion

**Comment by brianmhunt** [comment]
@grofit - Thanks for the report.

Is what you are thinking of similar to #662?

**Comment by grofit** [comment]
@brianmhunt Yeah seems pretty much the same, much like in the post I just augmented the existing foreach to provide the data as $iterator on the binding, so that way I could just do:

`bindingContext.$iterator.remove(bindingContext.$data);`

**Comment by mbest** [comment]
My [comment on #532](https://github.com/knockout/knockout/issues/532#issuecomment-11906160) describes the general problem with Knockout trying to provide the array in the context. I have a feeling that one of the things we'll be re-doing for version 4.0 (not for a while) is the `foreach` binding, at which time it should be simple to include this.
